src/image_downloader_lxd.rs

```diff
@@ -0,0 +1,311 @@
+use crate::image_downloader;
+use chrono::NaiveDateTime;
+use regex::Regex;
+use std::cmp::Ordering;
+use std::fs::{self, File};
+use std::io;
+use std::path::Path;
+use std::{error, fmt};
+
+const ROOTFS_SERVER_DOMAIN: &str = "https://us.lxd.images.canonical.com";
+const ROOTFS_FILE: &str = "rootfs.tar.xz";
+const ROOTFS_HASH_FILE: &str = "rootfs.tar.xz.asc";
+const IMAGE_META_URL: &str = "https://uk.lxd.images.canonical.com/meta/1.0/index-user";
+const ROOTFS: &str = "rootfs";
+
+#[derive(Debug)]
+pub enum Error {
+    ImageNotFound,
+    ImageMetaNotFound,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::ImageNotFound => write!(f, "Image Not Found"),
+            Error::ImageMetaNotFound => write!(f, "Image Meta Not Found"),
+        }
+    }
+}
+
+impl error::Error for Error {}
+
+pub struct Downloader {
+    /// ```https://us.lxd.images.canonical.com```から取得した
+    /// rootfsのメタデータ郡を ユーザが入力したディストリビューション、バージョンを用いてフィルタリングし、
+    /// メタデータを格納する
+    ///
+    /// # Exampld
+    /// ```ignore
+    /// [
+    ///     ImageMeta{
+    ///         distribution: "ubuntu",
+    ///         version: "focal",
+    ///         atch: "amd64",
+    ///         time: NaviveDateTime,
+    ///         path: "/images/ubuntu/focal/arm64/default/20220227_07:43/",
+    ///     },
+    ///     ImageMeta{
+    ///         distribution: "fedora",
+    ///         version: "34",
+    ///         atch: "amd64",
+    ///         time: NaviveDateTime,
+    ///         path: "/images/fedora/34/amd64/default/20220227_20:34/",
+    ///     },
+    /// ]
```

Comment:
ユーザーの指定した ディス鳥とバージョン に限定した，`ImageMeta`のベクタなら，
このように書いたほうが正確
```suggestion
    /// # Example
    /// ```ignore
    /// [
    ///     ImageMeta{
    ///         distribution: "ubuntu",
    ///         version: "focal",
    ///         atch: "amd64",
    ///         time: NaviveDateTime,
    ///         path: "/images/ubuntu/focal/arm64/default/20220227_07:43/",
    ///     },
    ///     ImageMeta{
    ///         distribution: "ubuntu",
    ///         version: "focal",
    ///         atch: "amd64",
    ///         time: NaviveDateTime,
    ///         path: "/images/ubuntu/focal/amd64/default/20220227_20:34/",
    ///     },
    /// ]
```